be2-scala/src/test/scala/ch/epfl/pop/pubsub/graph/validators/ElectionValidatorSuite.scala

```diff
@@ -630,4 +675,46 @@ class ElectionValidatorSuite extends TestKit(ActorSystem("electionValidatorTestA
     messageStandardActor shouldBe a[Left[PipelineError, _]]
     system.stop(dbActorRef.actorRef)
   }
+
+  test("Receiving the result of an election works as intended") {
+    val dbActorRef: AskableActorRef = mockDbForResultElection
+    val message: GraphMessage = new ElectionValidator(dbActorRef).validateResultElection(RESULT_ELECTION_RPC)
+    message should equal(Right(RESULT_ELECTION_RPC))
+    system.stop(dbActorRef.actorRef)
+  }
+
+  test("ResultElection with negative number of votes does not work in validateResultElection") {
+    val dbActorRef: AskableActorRef = mockDbForResultElection
+    val message: GraphMessage = new ElectionValidator(dbActorRef).validateResultElection(RESULT_ELECTION_RPC_WRONG)
+    message shouldBe a[Left[PipelineError, _]]
+    system.stop(dbActorRef.actorRef)
+  }
+
+  test("ResultElection with number of votes exceeding the number of attendees does not work in validateResultElection") {
+    val dbActorRef: AskableActorRef = mockDbForResultElection
+    val message: GraphMessage = new ElectionValidator(dbActorRef).validateResultElection(RESULT_ELECTION_RPC_TOO_MUCH_VOTES)
+    message shouldBe a[Left[PipelineError, _]]
+    system.stop(dbActorRef.actorRef)
+  }
+
+  test("ResultElection with election not ended does not work in validateResultElection ") {
+    val dbActorRef: AskableActorRef = mockDbElectionNotEnded
+    val message: GraphMessage = new ElectionValidator(dbActorRef).validateResultElection(RESULT_ELECTION_RPC)
+    message shouldBe a[Left[PipelineError, _]]
+    system.stop(dbActorRef.actorRef)
+  }
+
+  test("ResultElection with invalid question ids does not work in validateResultElection ") {
+    val dbActorRef: AskableActorRef = mockDbWrongQuestionIds
```

Comment:
I do not understand why you use the `mockDbWrongQuestionIds` instead of simply `mockDbForResultElection`. See my previous comments about this mockDb. 

You want to test the wrong Result message, thus no need to have a wrond mockDb.